catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -109,6 +133,72 @@ public void init() {
           res.header("Content-Encoding", "gzip");
         });
 
+    post(
+        "/cql/transform/:transformerId",
+        (req, res) -> {
+          String transformerId = req.params(":transformerId");
+          String body = util.safeGetBody(req);
+          CqlRequest cqlRequest = mapper.readValue(body, CqlRequest.class);
+          CqlQueryResponse cqlQueryResponse = executeCqlQuery(cqlRequest);
+
+          for (ServiceReference<QueryResponseTransformer> queryResponseTransformer :
+              queryResponseTransformers) {
+
+            LOGGER.trace("Finding transformer to transform query response.");
+
+            String id = (String) queryResponseTransformer.getProperty("id");
+
+            if (id.equals(transformerId)) {
+              BinaryContent content =
+                  bundleContext
+                      .getService(queryResponseTransformer)
+                      .transform(cqlQueryResponse.getQueryResponse(), new HashMap<>());
+
+              String mimeType = (String) queryResponseTransformer.getProperty("mime-type");
+              MimeTypes allTypes = MimeTypes.getDefaultMimeTypes();
+              String fileExt = allTypes.forName(mimeType).getExtension();
+
+              res.status(200);
+
+              String acceptEncoding = req.headers(HttpHeaders.ACCEPT_ENCODING);
+
+              boolean shouldGzip =
+                  StringUtils.isNotBlank(acceptEncoding)
+                      && acceptEncoding.toLowerCase().contains("gzip");
+
+              res.type(mimeType);
+              String attachment =
+                  String.format(
+                      "attachment;filename=export-%s%s", Instant.now().toString(), fileExt);
+              res.header("Content-Disposition", attachment);
+
+              if (shouldGzip) {
+                res.header(HttpHeaders.CONTENT_ENCODING, "gzip");
+                LOGGER.trace("Request header accepts gzip");
+              }
+
+              try (OutputStream servletOutputStream = res.raw().getOutputStream();
+                  InputStream resultStream = content.getInputStream()) {
+                if (shouldGzip) {
+                  try (OutputStream gzipServletOutputStream =
+                      new GZIPOutputStream(servletOutputStream)) {
+                    IOUtils.copy(resultStream, gzipServletOutputStream);
+                  }
+                } else {
+                  IOUtils.copy(resultStream, servletOutputStream);
+                }
+              }
+
+              LOGGER.trace("Response sent in {} file format.", fileExt);
+
+              return "";
```

Comment:
✏️ a `""` has no semantic meaning. Could you create a string variable with a name that describes this case?